### PR TITLE
Updates maploader code to be in line with tg's, startup improvements

### PR DESCRIPTION
**Pull request by Tastyfish**
- Ports /tg/'s version of the maploader library.
- Moves the away mission list to /config.
- Ports their map_template system, although it is currently only usable using the adminbus verbs.
 - Admins with +DEBUG can upload and add map templates to the world.
- Makes icon smoothing not use spawns, and instead, on startup, get smoothed by a startup hook.
- Also standardized some table deconstruction/attackby stuff that I noticed was a mess.
- Most relevant to most people, the changes appear to shave around 45 seconds off the world startup time after several tests.

:cl:
tweak: Startup time is now shorter.
rscdel: Removed the Station Collision away mission from rotation, due to multiple conceptual and technical issues.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
>Most relevant to most people, the changes appear to shave around 45 seconds off the world startup time after several tests.

Thank goodness.

File: config/example/away_mission_config.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #_maps/map_files/RandomZLevels/clownplanet.dmm
 #_maps/map_files/RandomZLevels/example.dmm
 #_maps/map_files/RandomZLevels/spacebattle.dmm
+#_maps/map_files/RandomZLevels/stationCollision.dmm
```
Comment (tigercat2000) [inline config/example/away_mission_config.txt]: uhhhhh

Comment (Tastyfish) [inline config/example/away_mission_config.txt]: I don't really have any intent on making that map better and in retrospect should not have readded it.

**Comment by KasparoVy** [comment]
Damn, that's hella fast

File: config/example/away_mission_config.txt (hunk 1 above)
Comment (Regen1) [inline config/example/away_mission_config.txt]: Not listed in the changelog?

Comment (Tastyfish) [inline config/example/away_mission_config.txt]: Oh right, sorry.

**Comment by Necaladun** [comment]
45 seconds of my life * rounds of SS13 I play = saving roughly 10 years

**Pull request merged**